src/studio/src/designer/frontend/packages/schema-editor/test/__tests__/schemaEditorSlice.test.ts

```diff
@@ -370,4 +370,91 @@ describe('SchemaEditorSlice', () => {
+      $ref: '#/definitions/Tekst_50',
+      displayName: 'ref',
+      path: "#/definitions/allOfTest/allOf/0",
+      groupItem: true,
+     };
+
+    // verify initial state => type is allOf
+    let item = state.uiSchema.find((f) => f.path === '#/definitions/allOfTest');
+    expect(item?.oneOf).toEqual(undefined);
+    expect(item?.allOf).toEqual([{ ...groupItemChild, path: '#/definitions/allOfTest/allOf/0'}]);
+    expect(item?.anyOf).toEqual(undefined);
+
+    const payload = {
+      path: '#/definitions/allOfTest',
+      type: 'oneOf',
+    };
+
+    // change to oneOf => verify changed state
+    let nextState = reducer(state, setGroupType(payload));
+    item = nextState.uiSchema.find((f) => f.path === '#/definitions/allOfTest');
+    expect(item?.oneOf).toEqual([{ ...groupItemChild, path: '#/definitions/allOfTest/oneOf/0'}]);
+    expect(item?.allOf).toEqual(undefined);
+    expect(item?.anyOf).toEqual(undefined);
+
+    // change to anyOf => verify changed state
+    nextState = reducer(state, setGroupType({
+      ...payload,
+      type: 'anyOf'
+    }));
+    item = nextState.uiSchema.find((f) => f.path === '#/definitions/allOfTest');
+    expect(item?.oneOf).toEqual(undefined);
+    expect(item?.allOf).toEqual(undefined);
+    expect(item?.anyOf).toEqual([{ ...groupItemChild, path: '#/definitions/allOfTest/anyOf/0'}]);
+
+    // change back to allOf => verify state
+    nextState = reducer(nextState, setGroupType({
+      ...payload,
+      type: 'allOf'
+    }));
+    item = nextState.uiSchema.find((f) => f.path === '#/definitions/allOfTest');
+    expect(item?.oneOf).toEqual(undefined);
+    expect(item?.allOf).toEqual([{ ...groupItemChild, path: '#/definitions/allOfTest/allOf/0'}]);
+    expect(item?.anyOf).toEqual(undefined);
+  });
+
+  it('handles deleting a "group" (anyOf, allOf, oneOf) child and shifting children paths', () => {
+    let item = state.uiSchema.find((f) => f.path === '#/definitions/anyOfTestSeveralItems');
+    expect(item?.anyOf?.length).toBe(4);
+    let nextState = reducer(state, deleteProperty({path: '#/definitions/anyOfTestSeveralItems/anyOf/1'}));
+    item = nextState.uiSchema.find((f) => f.path === '#/definitions/anyOfTestSeveralItems');
+    expect(item?.anyOf?.length).toBe(3);
+    expect(item?.anyOf?.[0].path).toBe('#/definitions/anyOfTestSeveralItems/anyOf/0');
+    expect(item?.anyOf?.[1].path).toBe('#/definitions/anyOfTestSeveralItems/anyOf/1');
+    expect(item?.anyOf?.[2].path).toBe('#/definitions/anyOfTestSeveralItems/anyOf/2');
+  });
+
+  it('handles adding child items to groups', () => {
+    // anyOf
+    let anyOfItem = state.uiSchema.find((f) => f.path === '#/definitions/anyOfTestSeveralItems');
+    expect(anyOfItem?.anyOf?.length).toBe(4);
+    let nextState = reducer(state, addGroupItem({ path: '#/definitions/anyOfTestSeveralItems', type: 'string'}));
+    let updatedAnyOfItem = nextState.uiSchema.find((f) => f.path === '#/definitions/anyOfTestSeveralItems');
```

Comment:
Consider using const when you never change the value of the variable later in the code. It will make it easier to review.